typescript/sdk/src/hook/HyperlaneHookDeployer.ts

```diff
@@ -0,0 +1,186 @@
+import debug from 'debug';
+
+import {
+  OptimismISM,
+  OptimismISM__factory,
+  OptimismMessageHook,
+  OptimismMessageHook__factory,
+  TestRecipient,
+  TestRecipient__factory,
+} from '@hyperlane-xyz/core';
+import { types } from '@hyperlane-xyz/utils';
+
+import { HyperlaneContracts, HyperlaneContractsMap } from '../contracts';
+import { HyperlaneDeployer } from '../deploy/HyperlaneDeployer';
+import { MultiProvider } from '../providers/MultiProvider';
+import { ChainMap, ChainName } from '../types';
+
+import { isHookConfig, isISMConfig } from './config';
+import { HookFactories, hookFactories } from './contracts';
+import { HookConfig } from './types';
+
+export class HyperlaneHookDeployer extends HyperlaneDeployer<
+  HookConfig,
+  HookFactories
+> {
+  constructor(multiProvider: MultiProvider) {
+    super(multiProvider, hookFactories, {
+      logger: debug('hyperlane:HookDeployer'),
+    });
+  }
+
+  async deploy(
+    configMap: ChainMap<HookConfig>,
+  ): Promise<HyperlaneContractsMap<HookFactories>> {
+    let ismContracts: HyperlaneContracts<HookFactories> | undefined;
+    let hookContracts: HyperlaneContracts<HookFactories> | undefined;
+
+    // Process ISM configs first
+    for (const [chain, config] of Object.entries(configMap)) {
+      if (isISMConfig(config)) {
+        ismContracts = await this.deployContracts(chain, config);
+      }
+    }
+
+    // Ensure ISM contracts have been deployed
+    if (!ismContracts) {
```

Comment:
Should we check that ISMContracts actually has keys too (i.e. isn't an empty object `{}`)?